src/mds/MDSRank.cc

```diff
@@ -2835,7 +2835,12 @@ void MDSRankDispatcher::handle_asok_command(
     command_dump_tree(cmdmap, *css, f);
   } else if (command == "dump loads") {
     std::lock_guard l(mds_lock);
-    r = balancer->dump_loads(f);
+    int64_t depth = -1;
```

Comment:
The limit is for a dirfrag rather than the entire tree. Is that what was intended?

Reply:
Without depth limit，mds command "dump loads" will output load info of dir and it's subdir in a breadth first manner until leaf. After setting depth limit, it will only output dir path no longer than depth. That's how it works for the entire tree in my opinion.

Reply:
ACK